Service/ExpressPay/Order/Update.php

```diff
@@ -85,9 +92,17 @@ public function execute($quoteMaskId, $gatewayId, $paypalOrderId): void
         }
 
         $websiteId = (int)$quote->getStore()->getWebsiteId();
-        $uri = "/checkout/orders/{{shopId}}/wallet_pay/$paypalOrderId";
+        $uri = "checkout/orders/{{shopId}}/wallet_pay/$paypalOrderId";
         $expressPayData = $this->quoteConverter->convertFullQuote($quote, $gatewayId);
 
+        $expressPayOrder = $this->getExpressPayOrder->execute($paypalOrderId, $gatewayId);
+        $expressPayOrderShipping = $expressPayOrder['data']['shipping_address'];
+        $hasShippingData = !empty($expressPayOrderShipping['country']) && !empty($expressPayOrderShipping['city']);
+
+        if (!$hasShippingData) {
+            unset($expressPayData['order_data']['shipping_address']);
```

Comment:
For express pay orders we don't have the shipping address on the create which means the patch will fail on update if we try to set it later. This adds checks in to avoid that so this endpoint can be used in both express and wallet pay scenarios.

Reply:
Fair enough, but the corresponding test needs to be updated to account for this.